src/SAML2/Assertion.php

```diff
@@ -1462,6 +1493,16 @@ private function addAttributeStatement(DOMElement $root)
                 $attribute->setAttribute('NameFormat', $this->nameFormat);
             }
 
+            if ($name === SAML2_Const::EPTI_URN_MACE || $name === SAML2_Const::EPTI_URN_OID) {
+                foreach ($values as $eptiValue) {
+                    $attributeValue = $document->createElementNS(SAML2_Const::NS_SAML, 'saml:AttributeValue');
+                    $attribute->appendChild($attributeValue);
+                    SAML2_Utils::addNameId($attributeValue, $eptiValue);
```

Comment:
@jaimeperez due to a broken hand I was not really capable of typing, hence a delayed response (hope to start work again tomorrow).

The proposed/implemented fixes are actually introducing potential bugs. As referenced in #76 , the value of an EPTI must be a NameID, and from #76 onward therefor expects the value to be a NameID compatible value. More specifically, that means at the very least an array with at least the key `Value` and as value of that key the actual value of the NameID. Therefor, allowing any arbitrary value as is done [here][1] is actually not conform the specification. Now, I'm not necessarily against the current implementation, but I do want to highlight that this allows for incorrect assertions to be created. 

[1]: https://github.com/simplesamlphp/saml2/blob/master/src/SAML2/Assertion.php#L1519-L1524